auth_backends/tests/test_backends.py

```diff
@@ -127,54 +126,41 @@ def test_login(self):
     @patch('auth_backends.backends.set_custom_attribute')
     @patch('auth_backends.backends.logger')
-    def test_start_with_session_cleanup(self, toggle_enabled, mock_logger, mock_set_attr):
-        """Test start method for session cleanup of existing user with toggle variation."""
-        with override_settings(ENABLE_OAUTH_SESSION_CLEANUP=toggle_enabled):
-            existing_user = User.objects.create_user(username='existing_user', email='existing@example.com')
+    def test_start_with_session_cleanup(self, user_authenticated, mock_logger, mock_set_attr):
+        """Test start method for session cleanup with and without authenticated user."""
+        request = RequestFactory().get('/auth/login/edx-oauth2/')
 
-            request = RequestFactory().get('/auth/login/edx-oauth2/')
+        if user_authenticated:
+            existing_user = User.objects.create_user(username='existing_user', email='existing@example.com')
             request.user = existing_user
 
-            middleware = SessionMiddleware(lambda req: None)
-            middleware.process_request(request)
-            request.session.save()
-
-            initial_session_key = request.session.session_key
-
-            self.backend.strategy.request = request
-
-            self.do_start()
+        middleware = SessionMiddleware(lambda req: None)
+        middleware.process_request(request)
+        request.session.save()
 
-            if toggle_enabled:
-                self.assertNotEqual(request.session.session_key, initial_session_key)
+        initial_session_key = request.session.session_key
 
-                self.assertTrue(request.user.is_anonymous)
+        self.backend.strategy.request = request
 
-                mock_set_attr.assert_has_calls([
-                    call('session_cleanup.toggle_enabled', True),
-                    call('session_cleanup.logout_performed', True),
-                    call('session_cleanup.logged_out_username', 'existing_user')
-                ], any_order=True)
+        self.do_start()
 
-                mock_logger.info.assert_called_with(
-                    "OAuth start: Performing session cleanup for user '%s'",
-                    'existing_user'
-                )
-            else:
-                self.assertEqual(request.session.session_key, initial_session_key)
+        if user_authenticated:
+            self.assertNotEqual(request.session.session_key, initial_session_key)
+            self.assertTrue(request.user.is_anonymous)
 
-                self.assertEqual(request.user, existing_user)
-                self.assertFalse(request.user.is_anonymous)
+            mock_set_attr.assert_called_once_with('session_cleanup.logout_required', True)
 
-                mock_set_attr.assert_has_calls([
-                    call('session_cleanup.toggle_enabled', False),
-                    call('session_cleanup.logout_performed', False)
-                ], any_order=True)
+            mock_logger.info.assert_called_with(
```

Comment:
I _think_ this is the only assertion needed in the conditional. You should be able to remove the `else` block, and move all other assertions after (or before) the conditional, with the caveat of updating one assertion to:
```
mock_set_attr.assert_called_once_with('session_cleanup.logout_required', user_authenticated)
```